Alexey Artishevskiy

dhvcc opened the pull request:
Modified gitignore to ignore PyCharm's .idea
Created a basic PyCharm project for FinalTask

dhvcc committed:
initial commit

Modified gitignore to ignore PyCharm's .idea
Created a basic PyCharm project for FinalTask

dhvcc committed:
Added argument parsing functionality with argparse

dhvcc committed:
Changed code to correspond PEP8 with 120 line length, not 79
Deleted default value for limit

dhvcc committed:
Packed everything in a rss_reader folder
Made rss reader a package(added __init__.py), added README.md

dhvcc committed:
Added url_parse.py which parses new and prints them out

dhvcc committed:
added logging functionality (--verbose)

dhvcc committed:
Added --json fucntionality, renamed url_parse.py to parsing.py

dhvcc committed:
Version 1.2.1. Configured package setup

dhvcc committed:
Updated README.md

dhvcc committed:
Updated README.md

dhvcc committed:
Merge branch 'FinalTask' of github.com:Kwizchm/PythonHomework into FinalTask

dhvcc committed:
Version 1.3.1. Added caching functionalty as required in iteration 3
Renamed README.md to README.rst and updated it

dhvcc committed:
Fixed version number

dhvcc committed:
Version 1.3.4. Made this package executable (renamed rss_reader.py to __main__.py)
Added --clear-cache argument
Added config module
Added CacheCleanError exception
Cleaned up code and modified some docstrings
Fixed minor bugs

dhvcc committed:
Version 1.4.0
Added raw converting functionality
Added various --to-* arguments
--to-mobi isn't working yet

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,34 @@
+    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+
+setup(
```
При установке через `pip install .` не подтягиваются зависимости 
```bash
# rss-reader --help
Traceback (most recent call last):
  File "/usr/local/bin/rss-reader", line 5, in <module>
    from rss_reader.__main__ import main
  File "/usr/local/lib/python3.8/site-packages/rss_reader/__main__.py", line 3, in <module>
    import rss_reader.news as news
  File "/usr/local/lib/python3.8/site-packages/rss_reader/news.py", line 2, in <module>
    from bs4 import BeautifulSoup
ModuleNotFoundError: No module named 'bs4'
```

dhvcc replied:
@AlexeiBuzuma Поправил всё в новой версии

AlexeiBuzuma commented on rss_reader/__main__.py:
```diff
@@ -0,0 +1,188 @@
+        log.info('There are no news from that date')
+    else:
+        print('Unknown error printing cached news')
+        exit(1)
```
По всему приложению стоит 26 вызовов функций `exit` 
На самом деле это очень много 
в идеальном случае точка выхода из программы должна быть одна 
в данном случае вместо принтов и вызова exit имеет смысл использовать исключения

AlexeiBuzuma commented on rss_reader/__main__.py:
```diff
@@ -0,0 +1,188 @@
+
+def main():
+    """Main function that is called when running a package"""
+    global cache_path, reader_dir
```
Это выглядит очень-очень подозрительно 
Использование глобальных переменных на запись -- это заранее не очень хорошая идея. 
Может быть есть смысл отрефакторить код так, чтобы не нужно было бы использовать глобальные переменные? 
или использовать ооп и атрибуты объекта для этих целей

AlexeiBuzuma commented on rss_reader/caching.py:
```diff
@@ -0,0 +1,143 @@
+    If there's no cache file, creates it and returns FileNotFoundError
+    Returns True if everything's fine"""
+    try:
+        f = open(cache_path, 'r+')
```
почему так открывается файл, без контекстного менеджера?

AlexeiBuzuma commented on rss_reader/caching.py:
```diff
@@ -0,0 +1,143 @@
+                    print(i['Description'])
+                    print()
+                    if 'Description links' in i:
+                        counter2 = 1
```
что такое `counter2` и для чего он нужен? :) 
очень странное название для переменной

dhvcc committed:
Version 1.4.1
Added functionality that allows user not to input source url when --clear-cache or --date DATE arguments are provided
Updated README, added cache location and cache json structure
Deleted --to-mobi argument
Cleaned up code
Fixed image mime-type detection

dhvcc committed:
Version 1.4.3
Fixed pip manual download in README
Added **kwargs in all functions
Added functionality that prevents invalid links
from getting in news dictionary
Main link is required by rss so it can still be broken

dhvcc committed:
Version 1.5.0
Added --colorize functionality
Updated README
Updated arguments, and their order, now they are prioritized from top to bottom

dhvcc committed:
Version 1.6.1
Fixed PR review issues
Got rid of all exit() calls, except one in main
Fixed install dependencies, now pip install works in venv
Updated error handling
Added new exceptions and provided all exceptions with error message

AlexeiBuzuma commented on requirements.txt:
```diff
@@ -0,0 +1,102 @@
+html5lib==1.0.1
+httplib2==0.9.2
+idna==2.8
+ipython==7.10.0
```
Все эти зависимости действительно нужны?

dhvcc replied:
@AlexeiBuzuma pip3 freeze их выдает, так что, думаю, да

AlexeiBuzuma commented on rss_reader/__main__.py:
```diff
@@ -0,0 +1,257 @@
+        try:
+            cached_news_dict = caching.get_cached_news(args.date, args.limit, cache_path)
+        except caching.NoNewsError as e:
+            raise e
```
Для чего тогда в целом нужен этот try except блок, если он ничего не обрабатывает, а просто бросает исключения дальше ?) 
фактически его можно удалить и ничего не изменится

AlexeiBuzuma commented on rss_reader/caching.py:
```diff
@@ -0,0 +1,102 @@
+    """Exception that is thrown when cache file doesn't exist"""
+
+    def __init__(self):
+        FileNotFoundError.__init__(self, 'Cache file does not exist')
```
Для этих задач обычно используют функцию `super()`

AlexeiBuzuma commented on rss_reader/converting.py:
```diff
@@ -0,0 +1,261 @@
+        raise Exception('Error converting to PDF: {}'.format(e))
+
+
+def to_fb2(news_dict, reader_dir, source, **kwargs):
```
Достаточно большая и сложная по своему содержанию функция. 
Советую ее разибть на несколько небольших

dhvcc committed:
Version 2.0.0
Added unit tests with 17% coverage
Deleted all comments
Cleaned up code
------------------------
Development discontinued

AlexeiBuzuma commented:
Моменты, которые можно улучшить: 
1) в `requirements.txt` лежит слишком много зависимостей, количество зависимостей можно уменьшить. 
2) функция `to_fb2` очень большая, скорее всего ее стоит разбить на несколько маленьких. 
3) в функциях fb2, to_pdf, to_ebub используется return код, который показывает успешность завершения функции. этого можно исбежать и просто генерировать исключения в местах, где происходит ошибка. 
4) вместо функции print советую использовать logging
5) я бы советовал использовать ООП для решения этой задачи. Этот подход в данном случае хорошо подходит.

dhvcc commented:
> Моменты, которые можно улучшить:
> 
> 1. в `requirements.txt` лежит слишком много зависимостей, количество зависимостей можно уменьшить.
> 2. функция `to_fb2` очень большая, скорее всего ее стоит разбить на несколько маленьких.
> 3. в функциях fb2, to_pdf, to_ebub используется return код, который показывает успешность завершения функции. этого можно исбежать и просто генерировать исключения в местах, где происходит ошибка.
> 4. вместо функции print советую использовать logging
> 5. я бы советовал использовать ООП для решения этой задачи. Этот подход в данном случае хорошо подходит.

1. Слишком поздно понял, после дедлайна
5. Конечно, почему-то не писал классы с самого начала и потом уже решил не переписывать
Спасибо :)
